common/formatting/token_partition_tree_test.cc

```diff
@@ -398,4 +398,486 @@ TEST_F(MoveLastLeafIntoPreviousSiblingTest, TwoGenerations) {
+
+  using tree_type = TokenPartitionTree;
+  tree_type tree{
+    all,
+    tree_type{one}, // header
+    tree_type{
+      args,
+      tree_type{two},
+      tree_type{three},
+      tree_type{four},
+      tree_type{five},
+      tree_type{six},
+    },
+  };
+
+  verible::BasicFormatStyle style;
+
+  // --------
+
+  {
+    UnwrappedLine group1(0, begin);
+    group1.SpanUpToToken(three.TokensRange().end());
+    UnwrappedLine group2(0, group1.TokensRange().end());
+    group2.SpanUpToToken(five.TokensRange().end());
+    UnwrappedLine group3(0, group2.TokensRange().end());
+    group3.SpanUpToToken(six.TokensRange().end());
+
+    tree_type unwrapped_tree{
+      all,
+      tree_type{group1,
+        tree_type{one},
+        tree_type{two},
+        tree_type{three},
+      },
+      tree_type{group2,
+        tree_type{four},
+        tree_type{five},
+      },
+      tree_type{group3,
+        tree_type{six},
+      },
+    };
+
+    style.column_limit = 12;
+    auto unwrapped_first(tree);
+    ReshapeFittingSubpartitions(unwrapped_first, style);
+    const auto diff = DeepEqual(unwrapped_first, unwrapped_tree, TokenRangeEqual);
+    EXPECT_TRUE(diff.left == nullptr) << "Expected:\n" << unwrapped_tree << "\n"
+                                                       << unwrapped_first << "\n";
+  }
+
+  // --------
+
+  {
+    UnwrappedLine group1(0, begin);
+    group1.SpanUpToToken(one.TokensRange().end());
+    UnwrappedLine group2(0, group1.TokensRange().end());
+    group2.SpanUpToToken(two.TokensRange().end());
+    UnwrappedLine group3(0, group2.TokensRange().end());
+    group3.SpanUpToToken(three.TokensRange().end());
+    UnwrappedLine group4(0, group3.TokensRange().end());
+    group4.SpanUpToToken(four.TokensRange().end());
+    UnwrappedLine group5(0, group4.TokensRange().end());
+    group5.SpanUpToToken(five.TokensRange().end());
+    UnwrappedLine group6(0, group5.TokensRange().end());
+    group6.SpanUpToToken(six.TokensRange().end());
+
+    tree_type wrapped_tree{
+      all,
+      tree_type{group1, tree_type{one}},
+      tree_type{group2, tree_type{two}},
+      tree_type{group3, tree_type{three}},
+      tree_type{group4, tree_type{four}},
+      tree_type{group5, tree_type{five}},
+      tree_type{group6, tree_type{six}},
+    };
+
+    style.column_limit = 5;
+    auto wrapped_first(tree);
+    ReshapeFittingSubpartitions(wrapped_first, style);
+    const auto diff = DeepEqual(wrapped_first, wrapped_tree, TokenRangeEqual);
+    EXPECT_TRUE(diff.left == nullptr) << "Expected:\n" << wrapped_tree << "\n"
+                                                       << wrapped_first << "\n";
+  }
+}
+
+// --------
+class ReshapeFittingSubpartitionsTestFixture : public ::testing::Test,
+                                               public UnwrappedLineMemoryHandler {
+ public:
+  ReshapeFittingSubpartitionsTestFixture()
+      : sample_("function_fffffffffff( "
+                "type_a_aaaa, type_b_bbbbb, "
```

Comment:
// combining what would normally be a type and a variable name into a single string for testing convenience
(if that is the reason)

Reply:
Yep. That's the reason. Added comment